src/controllers/krate/publish.rs

```diff
@@ -216,6 +216,16 @@ pub async fn publish(app: AppState, req: Parts, body: Body) -> AppResult<Json<Go
         AuthType::Regular(Box::new(auth))
     };
 
+    // Check if crate requires trusted publishing
+    if let Some(existing_crate) = &existing_crate
+        && existing_crate.trustpub_only
+        && matches!(auth, AuthType::Regular(_))
+    {
+        return Err(forbidden(
+            "You tried to publish with an API token but this crate requires trusted publishing.",
```

Comment:
I think people may be confused by "API token" as a term of art, so how about:

```suggestion
            "New versions of this crate can only be published using Trusted Publishing.",
```

Reply:
hmm... I specifically wanted to include the "API token" term to make it more obvious to the user what the problem is. I'm not sure what exactly you mean by "term of art". Do you think that crate owners that try to publish something are not aware of what the term "API token" means? 🤔

Reply:
Yeah, that's a good change. Makes it even more actionable. :+1: